src/components/label/LabelPrimitive.tsx

```diff
@@ -8,10 +8,12 @@ export interface LabelPrimitiveProps {
     className?: string;
     /** Text content of the label */
     labelContent: React.ReactNode;
+    /** Optional props for the label element */
+    labelElProps?: Object;
```

Comment:
Same here, we could simply extend `React.ComponentPropsWithoutRef<'label'>`. 

imo it's preferable to extend those and to add an explicit prop for the `span` element, instead of the other way around, since `label` is the outer/main element rendered by that component. In other words, from a consumer-perspective it is not clear, as of now, that everything else that you pass to `LabelPrimitive` component is forwarded to the inner `span` element.

Reply:
and what is the process for breaking changes, except that it should be indicated with prefix in commits and PR title? don't see anything else in documentation.

Reply:
Greg makes a lot of great points but I think you're right that it's a little tricky because of the possible impact. at the moment, we're trying to limit breaking changes since teams are encouraged to migrate to blueprint and so we don't want to cause concern if teams need to still upgrade to BUIE.

I'm not sure why the `{...rest}` was added to the `span` initially (the original implementation is from box-react-ui in 2016). Based on the explicit usages in `StandardLabel` and `HiddenLabel`, it shouldn't be doing anything lol. But I do see tooltips is utilizing the `...rest` props by adding additional props (`componentProps`) to the child component: https://github.com/box/box-ui-elements/blob/13ed6dadab00ba48127c0e3f72feb6b104f81f08/src/components/tooltip/Tooltip.tsx#L385
(not sure if there is an accessibility concern to move the attributes from the `span` to the `label` or if we need to try to keep the tooltip specific ones on the `span`)

up to you if you'd like to keep as-is or update it, just be cautious around the Tooltip use case. the process for breaking changes is just adding the prefix in the commit and PR, no other action needed

Reply:
I would leave as-is, and added more precise typing for `labelElProps` props, because we are migrating to `Blueprint` already.
If we decide that it is something that should be refactored, I think that we can create another task for that, and we also need to work on fix for running forms-client linked to BUIE, to test these changes. SWE from BUIE helped me to run that with workaround to test current fix.